### ISSUE TITLE
Booming Blade / Green Flame Blade not working

### ISSUE THREAD

**Issue by Blazewardog** [MrPrimate/ddb-importer#468]
**Describe the bug**
Both Booming Blade and Green Flame Bade aren't working.

I don't know much about the macro stuff in spells, but it looks like it is getting a copies the weapon and applies an effect and then you roll that. My guess is that the 3.0 5e system actually checks now that a weapon is in the character's inventory before you can attack with it.

Also looking at the callstack, the error is being generated in item.mjs in _onChatCardAction in this block which seems to support this:

```
      // Get the Item from stored flag data or by the item ID on the Actor
      const storedData = message.getFlag("dnd5e", "itemData");
      const item = storedData ? new this(storedData, {parent: actor}) : actor.items.get(card.dataset.itemId);
      if ( !item ) {
        ui.notifications.error(game.i18n.format("DND5E.ActionWarningNoItem", {
          item: card.dataset.itemId, name: actor.name
        }));
        return null;
      }
```

**To Reproduce**
1. Have a caster with either/or Green Flame Blade/Booming Blade
2. Cast either spell, pick one of your equipped weapons.
3. See that it ended up in the chat log
4. Click the Attack button
5. See a red error banner saying "The requested item no longer exists on Actor <charname>"

**Expected behavior**
A melee attack roll to happen.

**Screenshots**

**Environment:**
 - Browser Chrome
 - Foundry Version 11.315
 - 5e System Version 3.0.4
 - Module Version 4.0.16
 - Does this behaviour occur with other modules disabled: Yes

**Additional context**
An effected character you can import - https://www.dndbeyond.com/characters/112215903
Console Error - I cleared the console then triggered the error again
[requesteItemNoLongerExists.log](https://github.com/MrPrimate/ddb-importer/files/14580432/requesteItemNoLongerExists.log)

**Comment by MrPrimate** [comment]
It's working here. What versions of midi and DAE do you have installed?

Have you tried with just DDB Importer, Midi (and it's required modules)? No others?

**Comment by Blazewardog** [comment]
With just DDB-I + Midi (and its required modules) => I just got a template placer when casting GFB (no asking which weapon)

When I enabled enough to be allowed to check the DDB-I box for "Generate Automated Effects for spells" => I get the same error as my initial post.

Versions of things:
MidiQOL - 11.4.10 (I see there is a 11.4.11 from yesterday with a big "Breaking" on it, will try but tested this as-is)
DFred's - 6.0.3
DAE - 11.3.14

**Comment by Blazewardog** [comment]
I was playing around with MidiQOL settings, and found if I enabled "Auto Roll Attack" the macro works for Booming Blade. If that is how you are testing that could be the difference?

**Comment by MrPrimate** [comment]
Yep, that's it, I think I have fixed this in the next version, v5.0.1. Please can you check and let me know.

**Comment by Blazewardog** [comment]
Confirmed both Booming/Green Flame blade are working after reimporting the character.
